tools-ui/src/main/java/com/hedera/hashgraph/client/ui/popups/NewPasswordPopup.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Hedera Transaction Tool
+ *
+ * Copyright (C) 2018 - 2021 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.hashgraph.client.ui.popups;
+
+import com.codahale.passpol.BreachDatabase;
+import com.codahale.passpol.PasswordPolicy;
+import com.codahale.passpol.Status;
+import javafx.geometry.Insets;
+import javafx.geometry.Pos;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.PasswordField;
+import javafx.scene.input.KeyCode;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Priority;
+import javafx.scene.layout.VBox;
+import javafx.stage.Modality;
+import javafx.stage.Stage;
+
+import java.util.Arrays;
+
+import static com.hedera.hashgraph.client.core.constants.Constants.MAX_PASSWORD_LENGTH;
+import static com.hedera.hashgraph.client.core.constants.Constants.MIN_PASSWORD_LENGTH;
+import static com.hedera.hashgraph.client.core.constants.Constants.WHITE_BUTTON_STYLE;
+
+public class NewPasswordPopup {
+	private static char[] answer;
+
+	private NewPasswordPopup() {
+		throw new IllegalStateException("Utility class");
+	}
+
+	public static char[] display() {
```

Comment:
This method is maybe a bit long. 😀 

Refactoring as a sequence of calls to helper methods with descriptive  names would improve readability.

Reply:
Agreed. That was always my intention, but forgot to actually do it.
Done now =)